### PR TITLE
Stories/1044/html code prominent words

### LINKED ISSUE #1044: Html code appears in the prominent keyword section
Copied from https://github.com/Yoast/wordpress-seo-premium/issues/947

### What did you expect to happen?
Prominent Keywords To appear
### What happened instead?
Html output appeared instead. 

### With DIVI
**1.Add a post**

**2.Create using Divi's pagebuilder**

screen shot 2016-12-20 at 6 17 27 pm

**3.See content bloom html code appear in prominent keywords section**

screen shot 2016-12-20 at 6 17 34 pm

### With OptinMonster
**1.Add Optin Monster Shortcode**

screen shot 2017-02-23 at 10 28 36 am

**2.See fonts appear in Prominent Keywords Section**

screen shot 2017-02-23 at 10 28 56 am

**3.Remove Shortcode**

screen shot 2017-02-23 at 10 29 31 am

**4.See Prominent Keywords appear**

screen shot 2017-02-23 at 10 29 36 am

### What did you expect to happen?
Prominent keywords to appear
### What happened instead?
They did not
### How can we reproduce this behavior?
**1. Add a container, a text module, and a button**

screen shot 2017-02-22 at 10 27 40 am

**2. Edit the button. Go to the second tab called "Design". The first property is "Button Style". Set it to "Custom" and save.** 

**3.Update the page and wait for Yoast to analyse the content. Now check "Prominent Keywords" section.** 

screen shot 2017-02-22 at 10 27 19 am

**4.Edit the button again. Change button style to Green (or any other style) and save.** 

**5.Update the page and wait for Yoast to analyse the content.** 

**6.Now check "Prominent Keywords" section. It displays the right words.**

screen shot 2017-02-22 at 10 27 22 am

### Technical info
* WordPress version: 4.7.2
* Yoast SEO version: 4.3.2
* Fusion Builder: 1.0.6
*Fusion Core: 3.0.6
* Avada: 5.0.6

**Notes:** No JavaScript errors appear
-------

We should make sure we do not analyse the content within a `<style></style>`block.

### PR DESCRIPTION
**Pull request by CarolineGeven**
## Summary

This PR can be summarized in the following changelog entry:
Fixes a bug where style and script elements were parsed for the prominent words.

## Relevant technical choices:

* We've implemented a [htmlParser](https://github.com/fb55/htmlparser2) to parse the script tags.

## Test instructions

This PR can be tested by following these steps:

* In trunk, add style or script tags to your text. See that it gets picked up by the prominent words.
* Link this PR by doing a `yarn link` in the PR folder on your local environment. Link this PR to yoast seo premium by doing a `yarn link yoastseo` in your premium folder. See that the style/script tags do not get added to the prominent words anymore.

Fixes #1044

### CODE REVIEW

File: js/stringProcessing/htmlParser.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	 *
+	 * @param {string} tagName The tag name.
+	 *
+	 * @returns {void}
+	 */
+	onopentag: function( tagName ) {
+		if ( inlineTags.includes( tagName ) ) {
+			inScriptBlock = true;
+		} else {
+			textArray.push( "<" + tagName + ">" );
```
Comment (atimmer) [inline js/stringProcessing/htmlParser.js]: This will strip all attributes from all HTML tags. So our alt tag detection will be broken by this line.

Comment (atimmer) [inline js/stringProcessing/htmlParser.js]: When fixing this, also add a unit test for this behavior. When writing unit tests try to write tests that are outside the happy path.

File: spec/stringProcessing/htmlParserSpec.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	})
+	it( "filters out all script blocks", function() {
+		expect(htmlParser("Hi, this is a <script>Test</script>test.")).toEqual("Hi, this is a test.");
+	} )
+	it( "removes the script block when there is a type", function() {
+		expect( htmlParser( "<script type='text/javascript'>test</script>") ).toEqual("");
+	} )
+	it( "doesn't remove a div block", function() {
+		expect( htmlParser( "<div>Hello</div>" ) ).toEqual( "<div>Hello</div>" );
+	} )
```
Comment (atimmer) [inline spec/stringProcessing/htmlParserSpec.js]: I would add a test case where there is more variance in HTML tags. So combine a div with some attributes and a link and a style tag. Now only the style tag should be removed.

**Comment by atimmer** [comment]
CR Done 🚧

File: js/app.js [hunk 3]
```diff
@@ -627,7 +635,7 @@ App.prototype.showLoadingDialog = function() {
 /**
  * Updates the loading plugins. Uses the plugins as arguments to show which plugins are loading.
  *
- * @param   {Object}  plugins   The plugins to be parsed into the dialog.
+ * @param   {Object}  plugins   The plugins to be parsed into the dialog..
```
Comment (andizer) [inline js/app.js]: Please remove the extra full stop

File: js/stringProcessing/htmlParser.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	 * If the opening tag is not included in the inlineTags array, push the tag to the textArray.
+	 *
+	 * @param {string} tagName The tag name.
+	 * @param {object} nodeValue The attribute with the keys and values of the tag.
+	 * @returns {void}
+	 */
+	onopentag: function( tagName, nodeValue ) {
+		if ( inlineTags.includes( tagName ) ) {
+			inScriptBlock = true;
+		} else {
```
Comment (andizer) [inline js/stringProcessing/htmlParser.js]: If you do an early return in the if, the else isn't necessary and you can put the contents of the else block under the if.

```js 
if ( inlineTags.includes( tagName ) ) {
    inScriptBlock = true;
    
    return;
}

// The other code to execute.

File: js/stringProcessing/htmlParser.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+	 * If the closing tag is not included in the inlineTags array, push the tag to the textArray.
+	 *
+	 * @param {string} tagName The tag name.
+	 *
+	 * @returns {void}
+	 */
+	onclosetag: function( tagName ) {
+		if( inlineTags.includes( tagName ) ) {
+			inScriptBlock = false;
+		} else {
```
Comment (andizer) [inline js/stringProcessing/htmlParser.js]: Please do an early return to get rid of the else statement.

**Comment by andizer** [comment]
Cr Done 👍

**Pull request merged**